src/layers/conv.jl

```diff
@@ -1546,7 +1551,7 @@ function (l::SGConv)(g::AbstractGNNGraph, x,
     end
     c = 1 ./ sqrt.(d)
     for iter in 1:(l.k)
```

Comment:
I think that the fact that this layer iterates multiple times makes it unsuitable for heterographs operation. This PR should be closed, I'm sorry, but this operation just doesn't make sense for heterographs.

Reply:
@CarloLucibello does it mean we want to remove GNNHeteroGraph support from SGConv?
if yes, I can do that (and it's okay, it happens).
if no, in my opinion this PR still makes sense with some optimizations. 

I also noticed an oversight: the line 1338 should be amended from `x = x .* c'` to something that accounts for `x` being something else now. I thought I did it as per GCN, my mistake here.